sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.annotation.ServiceClientBuilder;
+
+/**
+ * builds table client
+ */
+@ServiceClientBuilder(serviceClients = {TableClient.class, TableAsyncClient.class})
+public class TableClientBuilder {
+    private String connectionString;
+    private String tableName;
+
+    /**
+     * Sets the connection string to help build the client
+     *
+     * @param connectionString the connection string to the storage account
+     * @return the TableClientBuilder
+     */
+    public TableClientBuilder connectionString(String connectionString) {
```

Comment:
The builder should also support either `TokenCredential` or `AzureKeyCredential`.